Lib/copy.py

```diff
@@ -98,23 +100,12 @@ def copy(x):
     return _reconstruct(x, None, *rv)
 
 
-_copy_dispatch = d = {}
-
-def _copy_immutable(x):
-    return x
-for t in (types.NoneType, int, float, bool, complex, str, tuple,
+_copy_atomic_types = {types.NoneType, int, float, bool, complex, str, tuple,
```

Comment:
Out of curiosity, would performance be better if we use a frozenset instead of a set? (and is it possible?)

Reply:
Good question, I'll benchmark a bit later. A frozenset should not require any locking, so perhaps there is a difference

Reply:
At this moment the `set` and `frozenset` have the same implementation for `__contains__`: 

https://github.com/python/cpython/blob/3bd7730bbda3e38db5920a7b8c95958ca90342bf/Objects/setobject.c#L2529-L2531

https://github.com/python/cpython/blob/3bd7730bbda3e38db5920a7b8c95958ca90342bf/Objects/setobject.c#L2416-L2420

so there is no performance difference. In the future however, for the free-threading build one could remove the critical section for the frozenset implementation here:

https://github.com/python/cpython/blob/3bd7730bbda3e38db5920a7b8c95958ca90342bf/Objects/setobject.c#L2198-L2207

Using a frozenset is possible, but this would add a bit of time to the import. On my system `%timeit frozenset(_copy_atomic_types)` is about 300 ns

Even faster than a `set`would be a data structure that looks only at the id of the objects involved (the `set` will use rich compare if no match is found, but that is not needed as all objects involved are singletons), but that is not available in cpython I believe.

Reply:
> but that is not available in cpython I believe.

That's right, it's not available.

---

Up to you if you want to make the free-threaded build faster in the future, but we should probably check the performances on this build. For now, let's keep the set for now (hopefully you'll rememeber this)